src/nnet3/nnet-test-utils.cc

```diff
@@ -888,13 +888,21 @@ void ComputeExampleComputationRequestSimple(
   request->outputs.clear();
   inputs->clear();
 
+  int32 t_modulus = GetTimePeriodForIvector(nnet);
   std::vector<Index> input_indexes, ivector_indexes, output_indexes;
   for (int32 n = n_offset; n < n_offset + num_examples; n++) {
     for (int32 t = input_start_frame; t < input_end_frame; t++)
       input_indexes.push_back(Index(n, t, 0));
     for (int32 t = output_start_frame; t < output_end_frame; t++)
       output_indexes.push_back(Index(n, t, 0));
-    ivector_indexes.push_back(Index(n, 0, 0));
+    if (t_modulus == 0) // case 1: single ivector for the entire chunk
```

Comment:
I don't think you answered my question. Why are you determining the the single ivector or multiple ivector case from the TimePeriodForIvector determined from the network ? This determination has to be made by the user. He/she decides this by either initializing ivector_ or online_ivectors_.